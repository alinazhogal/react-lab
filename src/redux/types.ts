import { Game, Platforms } from "@/components/home/games/games.types";

export const enum ActionsType {
  SET_SIGNIN_OPEN = "SET_SIGNIN_OPEN",
  LOGIN = "LOGIN",
  LOGOUT = "LOGOUT",
  SIGNUP = "SIGNUP",
  GET_USER = "GET_USER",
  CHANGE_PASSWORD = "CHANGE_PASSWORD",
  SAVE_PROFILE = "SAVE_PROFILE",
  GET_PROFILE = "GET_PROFILE",
  GET_TOP_GAMES = "GET_TOP_GAMES",
  SET_TOP_GAMES = "SET_TOP_GAMES",
  ADD_GAME = "ADD_GAME",
  UPDATE_GAME = "UPDATE_GAME",
  DELETE_GAME = "DELETE_GAME",
  GET_SEARCHED_GAMES = "GET_SEARCHED_GAMES",
  GET_FILTERED_GAMES = "GET_FILTERED_GAMES",
  SET_FILTERED_GAMES = "SET_FILTERED_GAMES",
  SET_SEARCH_LOADING = "SET_SEARCH_LOADING",
  GET_CART = "GET_CART",
  ADD_CART_ITEM = "ADD_CART_ITEM",
  UPDATE_CART_ITEM = "UPDATE_CART_ITEM",
  DELETE_CART_ITEM = "DELETE_CART_ITEM",
  BUY_ORDER = "BUY_ORDER",
  CLEAR_CART = "CLEAR_CART",
}

export interface ModalState {
  isOpen: boolean;
  callbackLink: string | undefined;
}

export interface SetSignInOpen {
  type: ActionsType.SET_SIGNIN_OPEN;
  payload: { isOpen: boolean; callbackLink: string };
}

export interface User {
  login: string;
  newLogin?: string;
  password?: string;
  newPassword?: string;
  phone?: string;
  description?: string;
  address?: string;
  photo?: string;
  role?: string;
  cart?: CartItem[];
  order?: CartItem[];
}

export interface AuthState {
  username: string;
  isAuth?: boolean;
  description?: string;
  phone?: string;
  address?: string;
  photo?: string;
  role?: string;
}

export interface Login {
  type: ActionsType.LOGIN;
  payload: {
    username: string;
    role: string;
    address: string;
    phone: string;
    description: string;
    photo: string;
  };
}

export interface SignUp {
  type: ActionsType.SIGNUP;
  payload: { username: string; role: string };
}

export interface Logout {
  type: ActionsType.LOGOUT;
}

export interface GetUser {
  type: ActionsType.GET_USER;
  payload: string;
}

export interface ChangePassword {
  type: ActionsType.CHANGE_PASSWORD;
}

export interface SaveProfile {
  type: ActionsType.SAVE_PROFILE;
  payload: {
    username: string;
    address: string;
    phone: string;
    description: string;
    photo: string;
  };
}

export interface GetProfile {
  type: ActionsType.GET_PROFILE;
  payload: {
    username: string;
    address: string;
    phone: string;
    description: string;
    photo: string;
  };
}
export interface GamesState {
  games: Game[];
  searchedGames: Game[] | undefined;
  isSearchLoading: boolean;
  isTopLoading: boolean;
  isFilterLoading: boolean;
}

export interface getTopGames {
  type: ActionsType.GET_TOP_GAMES;
}

export interface setTopGames {
  type: ActionsType.SET_TOP_GAMES;
  payload: Game[];
}

export interface getFilteredGames {
  type: ActionsType.GET_FILTERED_GAMES;
}

export interface setFilteredGames {
  type: ActionsType.SET_FILTERED_GAMES;
  payload: Game[];
}

export interface getSearchedGames {
  type: ActionsType.GET_SEARCHED_GAMES;
  payload: Game[] | undefined;
}

export interface setSearchedLoading {
  type: ActionsType.SET_SEARCH_LOADING;
  payload: boolean;
}

export interface AddGameCard {
  type: ActionsType.ADD_GAME;
  payload: Game;
}

export interface UpdateGameCard {
  type: ActionsType.UPDATE_GAME;
  payload: Game;
}
export interface DeleteGameCard {
  type: ActionsType.DELETE_GAME;
  payload: number;
}

export interface CartItem {
  id: number;
  name: string;
  platforms: Platforms[];
  selectedPlatform: Platforms;
  date: string;
  amount: number;
  price: number;
  image: string;
}

export interface GetCart {
  type: ActionsType.GET_CART;
  payload: CartItem[];
}

export interface AddCartItem {
  type: ActionsType.ADD_CART_ITEM;
  payload: CartItem;
}

export interface DeleteCartItem {
  type: ActionsType.DELETE_CART_ITEM;
  payload: string;
}

export interface BuyOrder {
  type: ActionsType.BUY_ORDER;
}

export interface ClearCart {
  type: ActionsType.CLEAR_CART;
}

export interface UpdateCartItem {
  type: ActionsType.UPDATE_CART_ITEM;
  payload: CartItem[];
}
