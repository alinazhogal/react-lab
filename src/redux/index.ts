import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import rootReducer from "./reducers/rootReducer";

// export const store = createStore(
//   rootReducer,
//   compose(
//     applyMiddleware(thunk),
//     window.__REDUX_DEVTOOLS_EXTENSION__ &&
//       window.__REDUX_DEVTOOLS_EXTENSION__({
//         serialize: true,
//       })
//   )
// );
export const store = createStore(rootReducer, applyMiddleware(thunk));

export type RootState = ReturnType<typeof store.getState>;
