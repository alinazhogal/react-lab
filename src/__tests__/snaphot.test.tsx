import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { store } from "../redux";
import ChangePassword from "../components/modal/forms/changePassword";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <ChangePassword onClose={() => 0} />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
