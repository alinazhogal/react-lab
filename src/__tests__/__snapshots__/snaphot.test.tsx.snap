// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders change password form correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="password"
      value=""
    />
    <span>
      
    </span>
  </div>
  <div>
    <label
      htmlFor="confirmPassword"
    >
      Repeat password
    </label>
    <input
      id="confirmPassword"
      name="confirmPassword"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="password"
      value=""
    />
    <span>
      
    </span>
  </div>
  <button
    className="button-el"
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`renders filters correctly 1`] = `
Array [
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <h4>
      Filters
    </h4>
    <img
      alt="arrow"
      className=""
      src="test-file-stub"
    />
  </div>,
  <div
    className="undefined"
  >
    <h3 />
    <div>
      <h5>
        Sort
      </h5>
      <div>
        <span>
          Criteria
        </span>
        <select
          id="criteria"
          name="sortCriteria"
          onChange={[Function]}
        >
          <option
            value="recent"
          >
            Recent
          </option>
          <option
            value="name"
          >
            Name
          </option>
          <option
            value="price"
          >
            Price
          </option>
        </select>
      </div>
      <div>
        <span>
          Type
        </span>
        <select
          id="type"
          name="sortType"
          onChange={[Function]}
        >
          <option
            value="asc"
          >
            Ascending
          </option>
          <option
            value="desc"
          >
            Descending
          </option>
        </select>
      </div>
    </div>
    <div>
      <h5>
        Age
      </h5>
      <label
        htmlFor="All ages"
      >
        <input
          checked={true}
          id="All ages"
          name="age"
          onChange={[Function]}
          type="radio"
          value="All ages"
        />
        All ages
      </label>
      <label
        htmlFor="3"
      >
        <input
          checked={false}
          id="3"
          name="age"
          onChange={[Function]}
          type="radio"
          value="3"
        />
        3+
      </label>
      <label
        htmlFor="6"
      >
        <input
          checked={false}
          id="6"
          name="age"
          onChange={[Function]}
          type="radio"
          value="6"
        />
        6+
      </label>
      <label
        htmlFor="12"
      >
        <input
          checked={false}
          id="12"
          name="age"
          onChange={[Function]}
          type="radio"
          value="12"
        />
        12+
      </label>
      <label
        htmlFor="18"
      >
        <input
          checked={false}
          id="18"
          name="age"
          onChange={[Function]}
          type="radio"
          value="18"
        />
        18+
      </label>
    </div>
    <div>
      <h5>
        Genres
      </h5>
      <label
        htmlFor="All genres"
      >
        <input
          checked={true}
          id="All genres"
          name="genre"
          onChange={[Function]}
          type="radio"
          value="All genres"
        />
        All genres
      </label>
      <label
        htmlFor="Shooter"
      >
        <input
          checked={false}
          id="Shooter"
          name="genre"
          onChange={[Function]}
          type="radio"
          value="Shooter"
        />
        Shooter
      </label>
      <label
        htmlFor="Arcade"
      >
        <input
          checked={false}
          id="Arcade"
          name="genre"
          onChange={[Function]}
          type="radio"
          value="Arcade"
        />
        Arcade
      </label>
      <label
        htmlFor="Survive"
      >
        <input
          checked={false}
          id="Survive"
          name="genre"
          onChange={[Function]}
          type="radio"
          value="Survive"
        />
        Survive
      </label>
    </div>
  </div>,
]
`;

exports[`renders profile page correctly 1`] = `
<section>
  <div
    className="section-content"
  >
    <h2>
      Profile page
    </h2>
    <div>
      <div>
        <div>
          <h4>
            No picture
          </h4>
        </div>
        <div
          role="button"
        >
          <label
            htmlFor="file"
          >
            Change photo
            <input
              accept="image/*"
              id="file"
              name="file"
              onChange={[Function]}
              type="file"
            />
          </label>
        </div>
        <button
          className="button-el"
          onClick={[Function]}
          type="button"
        >
          Change password
        </button>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <span>
            
          </span>
        </div>
        <div>
          <label
            htmlFor="phone"
          >
            Phone
          </label>
          <input
            id="phone"
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="tel"
            value=""
          />
          <span>
            
          </span>
        </div>
        <div>
          <label
            htmlFor="address"
          >
            Address
          </label>
          <input
            id="address"
            name="address"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <span>
            
          </span>
        </div>
        <div>
          <label
            htmlFor="desc"
          >
            Profile description
          </label>
          <textarea
            id="desc"
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            value=""
          />
          <span>
            
          </span>
        </div>
        <button
          className="button-el disabled"
          disabled={true}
          type="submit"
        >
          Save changes
        </button>
      </form>
    </div>
  </div>
</section>
`;
